### PR TITLE
Filename convention updated

### PR DESCRIPTION
**Pull request by shivamkm07**
Signed-off-by: shivam <shivamkm07@gmail.com>

# Description

This is for updating docker image file name to contain only the image and not have owner name as prefix.(e.g. dapr-1.7.0.tar.gz)

# Issue reference

Please reference the issue this PR will close: #2 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: .github/scripts/build_daprbundle.py [hunk 1]
```diff
@@ -120,12 +120,16 @@ def downloadBinaries(dir):
     unpack_archive(cli_filepath,dir)
     os.remove(cli_filepath)
 
+def getFileName(image):
```
Comment (mukundansundar) [inline .github/scripts/build_daprbundle.py:123]: add doc strings on what this message expects as input and what it outputs ,,, 

In general that can be done for any function ...

File: .github/scripts/build_daprbundle.py [hunk 2]
```diff
@@ -120,12 +120,16 @@ def downloadBinaries(dir):
...
+    imageBase = image.split("/")[-1]
```
Comment (shubham1172) [inline .github/scripts/build_daprbundle.py:124]: Can we add documentation for this function with example on what is expected?

File: .github/scripts/build_daprbundle.py (hunk 1 above)
Comment (shivamkm07) [reply .github/scripts/build_daprbundle.py:128]: Done!

File: README.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
 # Dapr Installer Bundle
-Dapr Installer Bundle contains CLI, runtime and dashboard packaged together. This eliminates the need to download binaries as well as docker images when initializing Dapr locally.
+
+## Overview
+Dapr Installer Bundle contains CLI, runtime and dashboard packaged together. This eliminates the need to download binaries as well as docker images when initializing Dapr locally, especially in airgap/offline environment. The bundle structure is fixed and is as follows:
+
+daprbundle
+├── dist
+│   ├── daprd_<runtime_os>_<runtime_arch>.tar.gz(`.zip` for windows)
```
Comment (shubham1172) [inline README.md:8]:
```suggestion
│   ├── daprd_<runtime_os>_<runtime_arch>.tar.gz (`.zip` for windows)
```

Comment (shubham1172) [reply README.md:8]: Same for below.

File: .github/scripts/build_daprbundle.py [hunk 4]
```diff
@@ -109,6 +112,7 @@ def downloadBinary(repo, fileBase, version, out_dir):
 
     print(f"Downloaded {url} to {downloadPath}")
 
+# Downloads all required dapr binaries(`daprd`,`placement`,`dashboard`) in `BIN_DIR` subfolder and dapr cli(`dapr`) in `dir` folder
```
Comment (shubham1172) [inline .github/scripts/build_daprbundle.py:115]: nit: space before parenthesis start, comma after `e.g.`, same for other places.

**Review by shubham1172 — APPROVED** [review]

File: .github/scripts/build_daprbundle.py [hunk 5]
```diff
@@ -120,12 +124,18 @@ def downloadBinaries(dir):
...
+# Downloads the givern version of docker image and saves it in `out_dir` folder
 def downloadDockerImage(image_name, version, out_dir):
     docker_image=f"{image_name}:{version}"
     if (version == "latest"):
         docker_image =image_name
-    fileName= docker_image.replace("/","-").replace(":","-") + ".tar.gz"
+    fileName = getFileName(docker_image)
     downloadPath = os.path.join(out_dir,fileName)
 
     if not os.path.exists(out_dir):
```
Comment (mukundansundar) [inline .github/scripts/build_daprbundle.py:141]: @shivamkm07 Once image is pulled, you have to retag it to a particular name before saving it ... 
The issue here is 
the image is pulled as `daprio/dapr:<version>.tar.gz`. 
When `docker save` is used to save an image to file ... the details about the repository, tag imageId, size, created at etc. all details are also saved as part of the tar.gz file. 

When docker load is used on the tar.gz file, the same image is loaded back with the same details. 

So even if the filename changes, the image name will still remain `daprio/dapr:<version>`.

For this one fix that I was able to do was once the image is pulled and before it is saved, 
run `docker tag daprio/dapr:<> dapr/dapr:<version>`. 

And then save that image.... `dapr/dapr:<version>`. 

Only then change we have the same name be used to again run the image. 

Consequently change the file name it is saved as `dapr-dapr-<version>.tar.gz`. As both the filename and image name needs to be one or that is what is assumed for now in CLI.

File: README.md (hunk 3 above)
Comment (shivamkm07) [reply README.md]: Done

File: .github/scripts/build_daprbundle.py [hunk 6]
```diff
@@ -120,12 +124,18 @@ def downloadBinaries(dir):
     unpack_archive(cli_filepath,dir)
     os.remove(cli_filepath)
 
+# Returns the fileName of the docker image to be saved. e.g. for image `daprio/dapr:1.7.0`, the fileName would be `dapr-1.7.0.tar.gz` i.e. without owner prefix
```
[the remark this answers is not in the document]
Comment (shivamkm07) [reply .github/scripts/build_daprbundle.py]: Done

File: .github/scripts/build_daprbundle.py (hunk 4 above)
Comment (shivamkm07) [reply .github/scripts/build_daprbundle.py]: Done

**Review by mukundansundar — APPROVED** [review]
lgtm

**Comment by mukundansundar** [comment]
@shivamkm07 Please fix DCO

**Review by mukundansundar — CHANGES_REQUESTED** [review]
dco fix please

**Comment by mukundansundar** [comment]
@shivamkm07 Make the details.json file read only by default ... 
Add comments In the file as to what each <key, value> represents ... 
Add a comment at the top of the file, cautioning users not to modify the file as it can result in undefined behavior.

**Review by artursouza — APPROVED** [review]

**Pull request merged**